lib/output/writer/blob_storage.go

```diff
@@ -0,0 +1,178 @@
+	if a.container, err = field.New(conf.Container); err != nil {
+		return nil, fmt.Errorf("failed to parse container expression: %v", err)
+	}
+	if a.path, err = field.New(conf.Path); err != nil {
+		return nil, fmt.Errorf("failed to parse path expression: %v", err)
+	}
+	if a.blobType, err = field.New(conf.BlobType); err != nil {
+		return nil, fmt.Errorf("failed to parse blob type expression: %v", err)
+	}
+	return a, nil
+}
+
+// ConnectWithContext attempts to establish a connection to the target Blob Storage Account.
+func (a *AzureBlobStorage) ConnectWithContext(ctx context.Context) error {
+	return a.Connect()
+}
+
+// Connect attempts to establish a connection to the target Blob Storage Account.
+func (a *AzureBlobStorage) Connect() error {
+	return nil
+}
+
+// Write attempts to write message contents to a target Azure Blob Storage container as files.
+func (a *AzureBlobStorage) Write(msg types.Message) error {
+	return a.WriteWithContext(context.Background(), msg)
+}
+
+func (a *AzureBlobStorage) getContainer(name string) (*azblob.ContainerURL, error) {
+	accountName, accountKey := a.conf.StorageAccount, a.conf.StorageAccessKey
+	if len(accountName) == 0 || len(accountKey) == 0 {
+		return nil, fmt.Errorf("invalid azure storage account credentials")
+	}
+	credential, err := azblob.NewSharedKeyCredential(accountName, accountKey)
+	if err != nil {
+		return nil, fmt.Errorf("invalid azure storage account credentials: %v", err)
+	}
+	p := azblob.NewPipeline(credential, azblob.PipelineOptions{})
+	URL, _ := url.Parse(fmt.Sprintf("https://%s.blob.core.windows.net/%s", accountName, name))
+	containerURL := azblob.NewContainerURL(*URL, p)
+	return &containerURL, nil
+}
+
+func (a *AzureBlobStorage) uploadToBlob(ctx context.Context, message []byte, blobName string, blobType string, containerURL *azblob.ContainerURL) error {
+	var err error
+
+	switch blobType {
+	case "BLOCK":
+		blobURL := containerURL.NewBlockBlobURL(blobName)
+		_, err = azblob.UploadStreamToBlockBlob(ctx, bytes.NewReader(message), blobURL, azblob.UploadStreamToBlockBlobOptions{})
+	case "APPEND":
+		blobURL := containerURL.NewAppendBlobURL(blobName)
+		_, err = blobURL.AppendBlock(ctx, bytes.NewReader(message), azblob.AppendBlobAccessConditions{}, nil)
+	}
+
+	return err
+}
+
+// WriteWithContext attempts to write message contents to a target storage account as files.
+func (a *AzureBlobStorage) WriteWithContext(wctx context.Context, msg types.Message) error {
+	ctx, cancel := context.WithTimeout(
+		wctx, a.timeout,
+	)
+	defer cancel()
+
+	return msg.Iter(func(i int, p types.Part) error {
+		c, err := a.getContainer(a.container.String(i, msg))
+		if err != nil {
+			return err
+		}
+		if err := a.uploadToBlob(ctx, p.Get(), a.path.String(i, msg), a.blobType.String(i, msg), c); err != nil {
+			a.log.Errorf("Error uploading blob: %v.", err)
+			if containerNotFound(err) {
+				if _, cerr := c.Create(ctx, azblob.Metadata{}, azblob.PublicAccessNone); cerr != nil {
```

Comment:
If this is successful would it make sense to attempt `uploadToBlob` once more?

Reply:
Agreed, retrying to upload to blob after the container is created successfully.